Ginger/Ginger/AutomatePageLib/NewAutomatePage.xaml.cs

```diff
@@ -1089,20 +1129,81 @@ private void xSaveBusinessFlowBtn_Click(object sender, RoutedEventArgs e)
                 }
             }
 
-            Reporter.ToStatus(eStatusMsgKey.SaveItem, null, mBusinessFlow.Name,
+            try
+            {
+                Reporter.ToStatus(eStatusMsgKey.SaveItem, null, mBusinessFlow.Name,
                                       GingerDicser.GetTermResValue(eTermResKey.BusinessFlow));
-            WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(mBusinessFlow);
-            Reporter.HideStatusMessage();
+                SwapLoadingPrefixText("Saving", false);
+
+                await WorkSpace.Instance.SolutionRepository.SaveRepositoryItemAsync(mBusinessFlow).ConfigureAwait(false);
+            }
+            finally
+            {
+                Reporter.HideStatusMessage();
+                SwapLoadingPrefixText("Saving", true);
+            }
+        }
+
+        public string loadingText { get; set; } = "Loading [BusinessFlow]...";
+
+        void SwapLoadingPrefixText(string swappedText, bool IsReset)
+        {
+            Dispatcher.Invoke(() =>
+            {
+                if (string.IsNullOrEmpty(swappedText))
+                {
+                    return;
+                }
+
+                if (IsReset)
+                {
+                    loadingText = loadingText.Replace(swappedText, "Loading");
+                    xItemsTabsSection.Visibility = Visibility.Visible;
+                    xItemsLoadingPnl.Visibility = Visibility.Collapsed;
+                }
+                else
+                {
+                    loadingText = loadingText.Replace("Loading", swappedText);
+                    xItemsTabsSection.Visibility = Visibility.Collapsed;
+                    xItemsLoadingPnl.Visibility = Visibility.Visible;
+                }
+
+                xLoadWindowText.Text = loadingText;
+                ToggleProcessButtons(IsReset);
+            });
         }
 
-        private void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
+        private async void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
         {
             if (CheckIfExecutionIsInProgress()) return;
 
-            if (Ginger.General.UndoChangesInRepositoryItem(mBusinessFlow, true))
+            if (mBusinessFlow.IsLocalBackupExist == false)
```

Comment:
Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5970337)